### PR TITLE
Do not ignore invalid command line options in JSON/XML interface [TG-9049]

### PR DESCRIPTION
**Pull request by peterschrammel**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [X] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by smowton — APPROVED** [review]

File: regression/cbmc/json-interface1/test_wrong_option.desc
Comment (majakusber) [inline regression/cbmc/json-interface1/test_wrong_option.desc]: :grey_question: What is this for?

File: regression/cbmc/json-interface1/test_wrong_flag.desc
Comment (majakusber) [inline regression/cbmc/json-interface1/test_wrong_flag.desc:1]: :grey_question: What is the `broken-smt-backend` for?

**Review by majakusber — APPROVED** [review]

**Comment by allredj** [comment]
@peterschrammel Is there a test-gen counterpart for that?

File: regression/cbmc/json-interface1/test_wrong_flag.desc
Comment (peterschrammel) [reply regression/cbmc/json-interface1/test_wrong_flag.desc:1]: The extra --cprover-smt2 flag would be ignored.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: dba23e12684205cc357ab3974055b95317768704).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/129741705

**Comment by majakusber** [comment]
TG bump: https://github.com/diffblue/test-gen/pull/3965

**Comment by majakusber** [comment]
I added a `do not merge` flag to make sure we wait for a passing tg bump.

**Comment by kroening** [comment]
This PR has the following issue: It gives `cmdlinet` a second interface, which is very different to the one it currently has. One uses a return value to report error, the other uses an exception.

Pick one, and use it for both use-cases.

**Comment by smowton** [comment]
@kroening suggest the `get` methods (that aren't optional, like `getoptnr`) should throw rather than ignore if there's no-such-option -- but that will no doubt raise more typo'd option cases to fix, so suggest we do that in a followup PR.

**Comment by smowton** [comment]
TG bump passed CI, can merge this once approved by @kroening

**Pull request merged**
